client/login/wp-login/test/test.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External Dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import { Login } from '../index.jsx';
+import useFakeDom from 'test/helpers/use-fake-dom';
+
+const defaultProps = {
+	magicLoginEnabled: true,
+	magicLoginView: '',
+	translate: ( str ) => {
+		return str;
+	}
+};
+
+describe( 'Login', () => {
+	useFakeDom();
+
+	describe( 'footerLinks', () => {
+		it( 'should not have a return button before the state is loaded', () => {
+			const login = new Login( defaultProps );
+			const links = login.footerLinks();
+
+			expect( links ).to.have.length( 2 );
+		} );
+
+		it( 'should not have a return button after the state is loaded, but there is no history', () => {
+			const login = new Login( defaultProps );
+			login.state.loaded = true;
```

Comment:
For testing react component we use `enzyme`, take a look at an example [here](https://github.com/Automattic/wp-calypso/blob/master/client/components/domains/domain-suggestion/test/index.js).

You can't just call the methods, because you need to emulate the whole react life-cycle.

For server side rendering you can use something like [that](https://github.com/Automattic/wp-calypso/blob/fcf789ebb43b7d8fe9a912d8fa82d5bee66573b5/server/pages/test/index.js#L31)